src/socket_server/deserialization.rs

```diff
@@ -0,0 +1,36 @@
+use crate::socket_server::frame::{Frame, Sound, Coord, Size, Sprite};
+use crate::coffee_time::output::frame::frame_generated::coffee_time::output::frame::root_as_frame;
+
+pub fn deserialize_frame(buffer: &[u8]) -> Result<Frame, String> {
+    let flat_frame = root_as_frame(buffer)
+        .map_err(|e| format!("Failed to get root as Frame: {:?}", e))?;
+
+    let sounds = flat_frame.sounds().ok_or_else(|| "No sounds found".to_string())?;
+    let sounds_vec: Vec<Sound> = sounds.iter().map(|sound| Sound {
+        can_play: sound.can_play(),
+        path: sound.path().unwrap_or_default().to_string(),
+    }).collect();
+
+    let sprites = flat_frame.sprites().ok_or_else(|| "No sprites found".to_string())?;
+    let sprites_vec: Vec<Sprite> = sprites.iter().map(|sprite| {
+        let position = sprite.position().ok_or_else(|| "Sprite position not found".to_string())?;
+        let size = sprite.size_().ok_or_else(|| "Sprite size not found".to_string())?;
+        Ok(Sprite {
+            position: Coord {
+                x: position.x(),
+                y: position.y(),
+            },
+            size: Size {
+                height: size.height(),
+                width: size.width(),
+            },
+            is_hidden: sprite.is_hidden(),
+            path: sprite.path().unwrap_or_default().to_string(),
```

Comment:
Well, I tried it, and I've been having problems with the buffer not living long enough.
For the moment, the solution is to use our own structures that manage that lifetime, as long as there are no efficiency or performance problems during the execution of the game, this can be maintained for the moment. I will consult with the POO to keep an eye on it.

Reply:
Yes, I agree. Maybe once a game is ready we will see it properly